holmes/plugins/toolsets/datadog/toolset_datadog_logs.py

```diff
@@ -0,0 +1,351 @@
+        )
+        logs.append(message)
+
+    return "\n".join(logs)
+
+
+class DatadogToolset(BasePodLoggingToolset):
+    dd_config: Optional[DatadogConfig] = None
+
+    def __init__(self):
+        super().__init__(
+            name="datadog/logs",
+            description="Toolset for interacting with Datadog to fetch logs",
+            docs_url="https://docs.datadoghq.com/api/latest/logs/",
+            icon_url="https://imgix.datadoghq.com//img/about/presskit/DDlogo.jpg",
+            prerequisites=[CallablePrerequisite(callable=self.prerequisites_callable)],
+            tools=[
+                PodLoggingTool(self),
+            ],
+            experimental=True,
+            tags=[ToolsetTag.CORE],
+        )
+
+    def fetch_pod_logs(self, params: FetchPodLogsParams) -> StructuredToolResult:
+        if not self.dd_config:
+            return StructuredToolResult(
+                status=ToolResultStatus.ERROR,
+                data=TOOLSET_CONFIG_MISSING_ERROR,
+                params=params.model_dump(),
+            )
+
+        try:
+            raw_logs = []
+            for storage_tier in self.dd_config.storage_tiers:
+                raw_logs = fetch_paginated_logs(
+                    params, self.dd_config, storage_tier=storage_tier
+                )
+
+                if raw_logs:
+                    # no need to try other storage tiers if the current one returned data
+                    break
+
+            if raw_logs:
+                logs_str = format_logs(raw_logs)
+                return StructuredToolResult(
+                    status=ToolResultStatus.SUCCESS,
+                    data=logs_str,
+                    params=params.model_dump(),
+                )
+            else:
+                return StructuredToolResult(
+                    status=ToolResultStatus.NO_DATA,
+                    params=params.model_dump(),
+                )
+
+        except DataDogRequestError as e:
+            logging.exception(e, exc_info=True)
+            return StructuredToolResult(
+                status=ToolResultStatus.ERROR,
+                error=f"Exception while querying Datadog: {str(e)}",
+                params=params.model_dump(),
+                invocation=json.dumps(e.payload),
+            )
+
+        except Exception as e:
+            logging.exception(
+                f"Failed to query Datadog logs for params: {params}", exc_info=True
+            )
+
+            return StructuredToolResult(
+                status=ToolResultStatus.ERROR,
+                error=f"Exception while querying Datadog: {str(e)}",
+                params=params.model_dump(),
+            )
+
+    def _perform_healthcheck(self) -> Tuple[bool, str]:
+        """
+        Perform a healthcheck by fetching a single log from Datadog.
+        Returns (success, error_message).
+        """
+        try:
+            logging.info("Performing Datadog configuration healthcheck...")
```

Comment:
This won't be good enough? https://docs.datadoghq.com/api/latest/authentication/#validate-api-key
We would like that we are able to indeed fetch logs?